src/CelesteTypes.jl

```diff
@@ -193,34 +205,14 @@ end
 # A vector of variational parameters.  The outer index is
 # of celestial objects, and the inner index is over individual
 # parameters for that object (referenced using ParamIndex).
+
 # TODO: use a matrix here, in conjunction with ArrayViews.jl (?)
+# TODO: Julia noob question -- is there a way to enforce the
+# differences between these variable types?  For now, this is
+# just helpful notation.
```

Comment:
Maybe, though that's more of Java-style solution: Julia only allows subtypes of abstract types. You could create a composite type that has just a single field, whose type is `Vector{Vector{Float64}}`. That would do it. It's also possible that enforcing the difference between `VariationalParams` and `Vector{Vector{Float64}}` wouldn't buy us much.